tests/Unit/Service/ShareServiceTest.php

```diff
@@ -0,0 +1,216 @@
+<?php
+/**
+ * Nextcloud - News
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author    Marco Nassabain <marco.nassabain@hotmail.com>
+ */
+
+
+namespace OCA\News\Tests\Unit\Service;
+
+use FeedIo\Explorer;
+use FeedIo\Reader\ReadErrorException;
+
+use OC\L10N\L10N;
+use OCA\News\Service\Exceptions\ServiceConflictException;
+use OCA\News\Service\Exceptions\ServiceNotFoundException;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCA\News\Service\FeedServiceV2;
+use OCA\News\Service\ItemServiceV2;
+use OCA\News\Service\ShareService;
+use OCA\News\Utility\Time;
+
+use OCA\News\Db\Feed;
+use OCA\News\Db\Item;
+use OCP\IConfig;
+use OCP\IL10N;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+
+
+class ShareServiceTest extends TestCase
+{
+    /**
+     * @var MockObject|ItemServiceV2
+     */
+    private $itemService;
+
+    /**
+     * @var MockObject|FeedServiceV2
+     */
+    private $feedService;
+
+    /**
+     * @var MockObject|LoggerInterface
+     */
+    private $logger;
+
+    /** @var ShareService */
+    private $class;
+
+    /**
+     * @var string
+     */
+    private $uid;
+
+    /**
+     * @var string
+     */
+    private $recipient;
+
+    protected function setUp(): void
+    {
+        $this->logger = $this->getMockBuilder(LoggerInterface::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->itemService = $this
+            ->getMockBuilder(ItemServiceV2::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->feedService = $this
+            ->getMockBuilder(FeedServiceV2::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->time = 333333;
+
+        $this->class = new ShareService(
+            $this->feedService,
+            $this->itemService,
+            $this->logger
+        );
+
+        $this->uid = 'sender';
+        $this->recipient = 'recipient';
+    }
+
+    public function testShareItemWithUser()
+    {
+        $feedUrl = 'http://nextcloud/sharedwithme';
+        $itemId = 3;
+
+        // Item to be shared
+        $item = new Item();
+        $item->setGuid('_guid_')
+            ->setGuidHash(md5('_guid_'))
+            ->setUrl('_url_')
+            ->setTitle('_title_')
+            ->setAuthor('_author_')
+            ->setUnread(0)
+            ->setStarred(1)
+            ->setFeedId(10);
+
+        // Shared item
+        $sharedItem = clone $item;
+        $sharedItem->setUnread(1)       // A newly shared item is unread, ...
+            ->setStarred(0)             // ... not starred, ...
+            ->setFeedId(100)            // ... placed in the 'Shared with me' feed, ...
+            ->setSharedBy($this->uid);  // ... and contains the senders user ID
+
+        // Dummy feed 'Shared with me'
+        $feed = new Feed();
```

Comment:
This should be a mock to make sure it's setting the correct values.

Reply:
Cannot mock it because we use `new Feed()`, and I cannot test it by testing the argument of the `feedService->insert` function - because the time field will be variable. I don't see a solution to test this, that's why the first time I left it like this